overlord/snapstate/autorefresh_gating_test.go

```diff
@@ -1270,7 +1270,7 @@ func fakeReadInfo(name string, si *snap.SideInfo) (*snap.Info, error) {
 	return info, nil
 }
 
-func (s *snapmgrTestSuite) TestAutoRefreshPhase2(c *C) {
+func (s *snapmgrTestSuite) testAutoRefreshPhase2(c *C, hold func(), expected []string) {
```

Comment:
nitpick: we tend to allow for nil same as nop for such test hooks

Reply:
also conceptually here hold is really about simulating the hook invocation? so maybe the func name should reflect that, and it just happens that the tests below hold in the hooks?

also do we need a test where instead the hooks would call proceed on something that was held instead?

Reply:
Done.